src/renderer/geometry/bounding_box.rs

```diff
@@ -47,40 +47,40 @@ impl BoundingBox {
         ];
 
         let rotations = vec![
-            Quat::zero(),
-            Quat::zero(),
-            Quat::zero(),
-            Quat::zero(),
-            Quat::from_angle_z(degrees(90.0)),
-            Quat::from_angle_z(degrees(90.0)),
-            Quat::from_angle_z(degrees(90.0)),
-            Quat::from_angle_z(degrees(90.0)),
-            Quat::from_angle_y(degrees(-90.0)),
-            Quat::from_angle_y(degrees(-90.0)),
-            Quat::from_angle_y(degrees(-90.0)),
-            Quat::from_angle_y(degrees(-90.0)),
+            Mat4::identity(),
+            Mat4::identity(),
+            Mat4::identity(),
+            Mat4::identity(),
+            Mat4::from_angle_z(degrees(90.0)),
+            Mat4::from_angle_z(degrees(90.0)),
+            Mat4::from_angle_z(degrees(90.0)),
+            Mat4::from_angle_z(degrees(90.0)),
+            Mat4::from_angle_y(degrees(-90.0)),
+            Mat4::from_angle_y(degrees(-90.0)),
+            Mat4::from_angle_y(degrees(-90.0)),
+            Mat4::from_angle_y(degrees(-90.0)),
         ];
 
         let scales = vec![
-            vec3(size.x, thickness, thickness),
-            vec3(size.x, thickness, thickness),
-            vec3(size.x, thickness, thickness),
-            vec3(size.x, thickness, thickness),
-            vec3(size.y, thickness, thickness),
-            vec3(size.y, thickness, thickness),
-            vec3(size.y, thickness, thickness),
-            vec3(size.y, thickness, thickness),
-            vec3(size.z, thickness, thickness),
-            vec3(size.z, thickness, thickness),
-            vec3(size.z, thickness, thickness),
-            vec3(size.z, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.x, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.x, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.x, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.x, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.y, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.y, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.y, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.y, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.z, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.z, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.z, thickness, thickness),
+            Mat4::from_nonuniform_scale(size.z, thickness, thickness),
```

Comment:
Since both `rotations` and `scales` now hold `Mat4`, you could combine the two vectors instead of doing the multiplication on line 84, doesn't matter much though.

Reply:
Yeah I got lazy 😄 I'll clean it up a bit.